src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -246,6 +249,15 @@ static int process_one_password_file(const char *filename, FILE *f) {
                         return 0;
                 }
 
+                if (arg_queryPID != 0) {
+                    if (pid != arg_queryPID /* TODO: && compare PID with arg_query */) {
+                        if (arg_action == ACTION_QUERY)
+                            log_info("Not querying '%s' (PID " PID_FMT "), ignored by query 'pid=%d'.", strna(message), pid, arg_queryPID);
+
+                        return 0;
+                    }
+                }
```

Comment:
coding style: please follow 8 space indenting

Reply:
why filter this late? seems like you should filter much earlier on, i.e. the moment the file is parsed

Reply:
moved earlier, just after basic checks